internal/common/slices/slices.go

```diff
@@ -159,6 +170,32 @@ func Subtract[T comparable](list []T, toRemove []T) []T {
 	return out
 }
 
+// ParallelFilter is semantically the same as Filter except that results can be returned out of order
+func ParallelFilter[S ~[]E, E any](s S, predicate func(e E) bool) S {
```

Comment:
Do we have evidence that this is needed (e.g. it seems fairly marginal for your use case of filtering queues).  If not I'd just remove this as there's a new streams api coming to GO so imagine we'll want to deprecate most of the code here.